File: app/src/main/java/org/dhis2/data/database/SqlBriteDatabaseAdapter.java
package org.dhis2.data.database;

import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteStatement;
import androidx.annotation.NonNull;

import com.squareup.sqlbrite2.BriteDatabase;

import org.hisp.dhis.android.core.arch.db.access.DatabaseAdapter;
import org.hisp.dhis.android.core.arch.db.access.Transaction;

import static org.dhis2.utils.Preconditions.isNull;

class SqlBriteDatabaseAdapter implements DatabaseAdapter {
    private final BriteDatabase sqlBriteDatabase;

    SqlBriteDatabaseAdapter(@NonNull BriteDatabase briteDatabase) {
        isNull(briteDatabase, "briteDatabase == null");
        sqlBriteDatabase = briteDatabase;
    }

    @Override
    public SQLiteStatement compileStatement(String sql) {
        return sqlBriteDatabase.getWritableDatabase().compileStatement(sql);
    }

    @Override
    public Cursor query(String sql, String... selectionArgs) {
        return sqlBriteDatabase.query(sql, selectionArgs);
    }

    @Override
    public long executeInsert(String table, SQLiteStatement sqLiteStatement) {
        return sqlBriteDatabase.executeInsert(table, sqLiteStatement);
    }

    @Override
    public int executeUpdateDelete(String table, SQLiteStatement sqLiteStatement) {
        return sqlBriteDatabase.executeUpdateDelete(table, sqLiteStatement);
    }

    @Override
    public int delete(String table, String whereClause, String[] whereArgs) {
        return sqlBriteDatabase.delete(table, whereClause, whereArgs);
    }

    @Override
    public int delete(String table) {
        return sqlBriteDatabase.delete(table, null);
    }

    @Override
    public Transaction beginNewTransaction() {
        return new SqlBriteTransaction(sqlBriteDatabase.newTransaction());
    }

    @Override
    public SQLiteDatabase database() {
        return sqlBriteDatabase.getReadableDatabase();
    }
}

File: app/src/main/java/org/dhis2/data/database/SqlBriteDatabaseAdapter.kt
package org.dhis2.data.database

import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteStatement

import com.squareup.sqlbrite2.BriteDatabase

import org.dhis2.utils.Preconditions
import org.hisp.dhis.android.core.data.database.DatabaseAdapter
import org.hisp.dhis.android.core.data.database.Transaction


internal class SqlBriteDatabaseAdapter(private val sqlBriteDatabase: BriteDatabase) : DatabaseAdapter {

    init {
        Preconditions.isNull(sqlBriteDatabase, "briteDatabase == null")
    }

    override fun compileStatement(sql: String): SQLiteStatement {
        return sqlBriteDatabase.writableDatabase.compileStatement(sql)
    }

    override fun query(sql: String, vararg selectionArgs: String): Cursor {
        return sqlBriteDatabase.query(sql, *selectionArgs)
    }

    override fun executeInsert(table: String, sqLiteStatement: SQLiteStatement): Long {
        return sqlBriteDatabase.executeInsert(table, sqLiteStatement)
    }

    override fun executeUpdateDelete(table: String, sqLiteStatement: SQLiteStatement): Int {
        return sqlBriteDatabase.executeUpdateDelete(table, sqLiteStatement)
    }

    override fun delete(table: String, whereClause: String, whereArgs: Array<String>): Int {
        return sqlBriteDatabase.delete(table, whereClause, *whereArgs)
    }

    override fun delete(table: String): Int {
        return sqlBriteDatabase.delete(table, null)
    }

    override fun beginNewTransaction(): Transaction {
        return SqlBriteTransaction(sqlBriteDatabase.newTransaction())
    }

    override fun database(): SQLiteDatabase {
        return sqlBriteDatabase.readableDatabase
    }
}
